src/app/qgsloadstylefromdbdialog.cpp

```diff
@@ -102,14 +107,98 @@ QString QgsLoadStyleFromDBDialog::getSelectedStyleId()
   return mSelectedStyleId;
 }
 
-void QgsLoadStyleFromDBDialog::cellSelectedRelatedTable( int r )
+void QgsLoadStyleFromDBDialog::setLayer( QgsVectorLayer *l )
 {
-  mLoadButton->setEnabled( true );
-  mSelectedStyleId = mRelatedTable->item( r, 0 )->data( Qt::UserRole ).toString();
+  mLayer = l;
+  if ( mLayer->dataProvider()->isDeleteStyleFromDBSupported() )
+  {
+    //QgsDebugMsg( "QgsLoadStyleFromDBDialog::setLayer → The dataProvider supports isDeleteStyleFromDBSupported" );
+    mDeleteButton->setVisible( true );
+  }
+  else
+  {
+    // QgsDebugMsg( "QgsLoadStyleFromDBDialog::setLayer → The dataProvider does not supports isDeleteStyleFromDBSupported" );
+    mDeleteButton->setVisible( false );
+  }
+}
+
+void QgsLoadStyleFromDBDialog::relatedTableSelectionChanged()
+{
+  selectionChanged( mRelatedTable );
+  //deselect any other row on the other table widget
+  QTableWidgetSelectionRange range( 0, 0, mOthersTable->rowCount() - 1, mOthersTable->columnCount() - 1 );
+  mOthersTable->setRangeSelected( range, false );
 }
 
-void QgsLoadStyleFromDBDialog::cellSelectedOthersTable( int r )
+void QgsLoadStyleFromDBDialog::otherTableSelectionChanged()
 {
-  mLoadButton->setEnabled( true );
-  mSelectedStyleId = mOthersTable->item( r, 0 )->data( Qt::UserRole ).toString();
+  selectionChanged( mOthersTable );
+  //deselect any other row on the other table widget
+  QTableWidgetSelectionRange range( 0, 0, mRelatedTable->rowCount() - 1, mRelatedTable->columnCount() - 1 );
+  mRelatedTable->setRangeSelected( range, false );
+}
+
+void QgsLoadStyleFromDBDialog::selectionChanged( QTableWidget *styleTable )
+{
+  QTableWidgetItem *item;
+  QList<QTableWidgetItem *> selected = styleTable->selectedItems();
+
+  if ( selected.count() > 0 )
```

Comment:
!selected.isEmpty()

(Using isEmpty makes the meaning of this check immediately obvious)